langchain4j-bedrock/src/main/java/dev/langchain4j/model/bedrock/BedrockTitanEmbeddingModel.java

```diff
@@ -40,37 +40,35 @@ protected String getModelId() {
     /**
      * 1024 is default size of output vector for Titan Embedding model V2.
      */
-    @Builder.Default
-    private final Integer dimensions = 1024;
+    private final Integer dimensions;
 
     /**
-     *  A flag indicating whether or not to normalize the output embeddings.
+     *  A flag indicating whether to normalize the output embeddings.
      *  It defaults to true, which is optimal for RAG use cases.
      */
-    @Builder.Default
-    private final Boolean normalize = true;
+    private final Boolean normalize;
 
 
     @Override
     protected List<Map<String, Object>> getRequestParameters(List<TextSegment> textSegments) {
         if (MODEL_V1_ID.equals(this.model)) {
+            if (this.dimensions != null || this.normalize != null) {
+                throw new IllegalArgumentException("Dimensions and normalize are not supported for Titan Embedding model V1");
```

Comment:
I would even remove this check (and check for `MODEL_V1_ID`) and just pass through all parameters that user set to the AWS and let it decide if it is legal or not.

I can imagine at some point `MODEL_V1_ID` might allow setting this parameters and the logic on our side will eb outdated.

Reply:
If we remove this check too, then it will fail at AWS side. Remember?
`Malformed input request: 2 schema violations found, please reformat your input and try again. (Service: BedrockRuntime, Status Code: 400, Request ID: b606929b-e642-4c34-8565-bc4e32f66510)`

And one of your comments was that we should fail on "our" (langchain4j) side.

So I'm OK to implement either way, just let's decide.
Thank You

Reply:
Any news on this? How do you want to proceed?

Reply:
Let's keep it as it is now, thank you!